### ISSUE TITLE
store.findAll not refetching after a removeAll unless force parameters is set

### ISSUE THREAD

**Issue by Goopil** [js-data/js-data#451]
Hello, 

this is more of a question than a issue.

In the case that i want to prune a specific resource from the store. I have to call removeAll(resourceName) now right ? so why on the next findAll on that resource, the store dosen't refetch the collection via the adapter unless i force it to it ? 

if i understand js-data correctly, the store cache the query to not redo the same query ? 

so my question is removeAll() dosen't unregister previously computed request ? Or is it the empty query that dosen't get unregistered ? 

you can see it [there](https://plnkr.co/edit/HTy80eaXRinWmTbWRu8U)

**Comment by jmdobry** [comment]
This is bug. Until I release a fix, you can do this:

```js
// Clear a single query:
store._completedQueries[mapperName][store.hashQuery(mapperName, query)] = undefined;
// Clear all queries:
store._completedQueries[mapperName] = {};
```

**Comment by jmdobry** [comment]
Thanks for catching this and providing the plunker for reproducing the issue!

**Comment by Goopil** [comment]
thanks for your quick response. And above all thanks for the great tool. I really enjoy working with it.

this work for me just in case. (plunk updated)
```
store.on('remove', function (name, collection) {
  if (store.getAll(name) < 1) {
    store._completedQueries[name][store.hashQuery(name)] = undefined;
  }
})
```
There is a little something that come to mind. Shouldn't there be the standard before / after events remove/ removeAll  cicle ?

**Comment by jmdobry** [comment]
> Shouldn't there be the standard before / after events on remove / removeAll methods ?

No, just like there's no before/after add/change.

These events are not part of the Mapper's CRUD interface, they are artifact's of JSData's other in-memory components, such as the `Collection` and `DataStore` classes.

**Comment by esase** [comment]
@jmdobry hello, when you are going to fix this and make next release?

**Comment by Goopil** [comment]
You're right, this was more a suggestion to have a consistant hook api on each methods ;)

**Comment by esase** [comment]
@jmdobry  I also found a bug: 
I get all items like this: `this.api.findAll('users', {status: 'active});`
and it works only once!
If you will try to call it again: `this.api.findAll('users', {status: 'active});` you will get empty array. There is no cached items and it doesn't load data again from adapter. Just gets an empty array

**Comment by Goopil** [comment]
even with `this.api.findAll('users', { status: 'active' }, { force:true });` or `this.api.getAll('users')`  ?

**Comment by esase** [comment]
@Goopil No I didn't use the force parameter. I don't understand why the method returns an empty array? What the logic implemented there. If I think correctly the method should check my second request inside cached items and returns previously loaded data. if there is no any cached items load them again. I'm I right?

**Comment by prionkor** [comment]
Does anyone have a solution for this? This makes whole library unusable. :(

**Comment by jmdobry** [comment]
So, the original bug reported at the top of this issue (which was previously demonstrated [here](https://plnkr.co/edit/HTy80eaXRinWmTbWRu8U?p=preview)) appears to be fixed. @Goopil Correct me if I'm mistaken, but when I try that plunker with js-data 3.0.1 and js-data-http 3.0.0 your error case is not triggered, so I'm going to consider this fixed.

If someone finds a similar problem please open another issue with a plunker or jsfiddle or something demonstrating the issue.
